### PR TITLE
[NO GBP] Fixes even more AI related CI runtimes

### PR DESCRIPTION
**Pull request by vinylspiders**
## About The Pull Request

Consider this a continuation of https://github.com/tgstation/tgstation/pull/80202

firefox_P62DdMv946

~~It seems I missed a few.~~

Edit: Modified per request to handle this more broadly. If a pawn gets `qdel`'d, the ai controller should be set to off and get removed from the list of active controllers, and all their actions should be canceled.

Also adds some qdeleted checks to `finish_action()`, which can still run after the pawn gets qdeleted as part of the `CancelActions()` chain.

## Why It's Good For The Game

Less spurious CI failures.

## Changelog

Nothing player facing really.

### CODE REVIEW

File: code/datums/ai/babies/babies_subtrees.dm [hunk 1]
```diff
@@ -18,7 +18,7 @@
 		controller.queue_behavior(reproduce_behavior, BB_BABIES_TARGET, BB_BABIES_CHILD_TYPES)
 		return SUBTREE_RETURN_FINISH_PLANNING
 
-	if(controller.pawn.gender == FEMALE || !controller.blackboard[BB_BREED_READY])
+	if(QDELETED(controller.pawn) || controller.pawn.gender == FEMALE || !controller.blackboard[BB_BREED_READY])
```
Comment (Jacquerel) [inline code/datums/ai/babies/babies_subtrees.dm:21]: It seems very silly to add this individually to a bunch of subtrees as opposed to just moving this up one layer to "before we call `SelectBehaviours` on any subtree"
Subtrees _should_ just be queueing rather than executing actions if I remember the flow correctly, so if you're deleted at any step here you'll be deleted in all of them and it doesn't make sense to do any planning.

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/basic/pets/parrot/parrot_ai/parrot_perching.dm [hunk 2]
```diff
@@ -8,13 +8,10 @@
...
 	var/mob/living/living_pawn = controller.pawn
-	if(QDELETED(living_pawn))
-		return
-
-	var/atom/buckled_to = living_pawn.buckled
+	var/atom/buckled_too = living_pawn.buckled
 
 	//do we have a current target or is chance to unbuckle has passed? then unbuckle!
-	if(buckled_to)
+	if(buckled_too)
```
Comment (Jacquerel) [inline code/modules/mob/living/basic/pets/parrot/parrot_ai/parrot_perching.dm:14]: Why did you change this? The former name was correct, now it is wrong.

File: code/datums/ai/babies/babies_subtrees.dm (hunk 1 above)
Comment (Jacquerel) [reply code/datums/ai/babies/babies_subtrees.dm]: Sorry for not replying to this I forgot
but yeah what you have done in these later commits looks like the right approach to me, there's no reason AI behaviours should continue planning or running when the mob is deleted

File: code/modules/mob/living/basic/pets/parrot/parrot_ai/parrot_perching.dm (hunk 2 above)
Comment (Jacquerel) [reply code/modules/mob/living/basic/pets/parrot/parrot_ai/parrot_perching.dm:14]: It is "buckled to" as in "the thing I am buckled to"

**Comment by ZephyrTFA** [comment]
I hate AI. I hate AI.

**Comment by Ben10Omintrix** [comment]
> @Ben10Omintrix do you have any idea what might be causing these?
> 
> image
> 
> They're the only ones that are somehow still getting their `pawn` nulled midway through planning. Which shouldn't be happening but I couldn't tell exactly what was causing it...

i think the do after might be upsetting it for some reason. im going to try to completely pause the AI during do afters and see if that solves anything
